x/metadata/keeper/signers.go

```diff
@@ -338,29 +352,81 @@ func (k Keeper) validateProvenanceRole(ctx sdk.Context, parties []*PartyDetails)
 	return nil
 }
 
+// isWasmAccount returns true if the provided addr is the address of a smart contract account.
+// A smart contract account is a BaseAccount that exists, has a sequence of 0 and does not have a public key.
+func (k Keeper) isWasmAccount(ctx sdk.Context, addr sdk.AccAddress) bool {
```

Comment:
A marker address wouldn't return `true` because it's not a `BaseAccount`. However, something like the gov module account would return true. I'm not sure of another way to test if it's a smart contract though. In the wasm module, this same check is done. I can't remember if they limit it to only `BaseAccount`s though.

I guess, if a false positive starts causing problems, we can look deeper into it. Last I knew, the wasm module didn't have an "is this a smart contract address" function. We could try getting it, but that's a whole lot of reading that isn't needed.